### ISSUE TITLE
switch-case-fall-through check failed on blocked case clauses

### ISSUE THREAD

**Issue by falsandtru** [palantir/tslint#785]
``` ts
switch (0) {
  case 0: {
    break;
  }
  default:
}
```

**Comment by JKillian** [comment]
Thanks for the report @falsandtru. For a possible workaround, I'd try using this style for now:

```typescript
switch (0) {
  case 0: {
    //whatever code here
  } break;
  default:
}
```

**Comment by falsandtru** [comment]
Thanks for your response. But I cannot solve some cases.

```ts
switch (n) {
  case 0: {
    let m = n;
    // ...
    return n * m;
  }
  default:
}
```

Would you fix this issue?

**Comment by JKillian** [comment]
@falsandtru Yes, this should definitely be fixed.
Also, @Think7 also encountered this problem over in #898.

**Comment by nsgundy** [comment]
Just ran into the same issue, so I am all for having it fixed :)

Thanks for making tslint :+1: 

JSHint used to have the same problem, see https://github.com/jshint/jshint/issues/858.

**Comment by JKillian** [comment]
Wanted to mention that the newest versions of the TS compiler now have a [--noFallthroughCasesInSwitch](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#case-clause-fall-throughs) flag which could lead to this rule being deprecated soon.

**Comment by falsandtru** [comment]
I agree, this rule will be needless.

**Comment by JKillian** [comment]
Although a TSLint rule offers the advantage that you can selectively disable it, whereas as far as I know you can't do that with the compiler flag.

**Comment by falsandtru** [comment]
Then TSLint would be better to change its default(example) config to disable.

**Comment by 0815fox** [comment]
Yeah, actually the `--noFallthroughCasesInSwitch` - Option is even more dangerous, because you will not notice, that it does not fall through even though you expected it to. I would prefer to have this tslint option fixed and not use `--noFallthroughCasesInSwitch`.

**Comment by ufukomer** [comment]
Fix this pls!
+1

**Comment by kuldarim** [comment]
+1

**Comment by Jan539** [comment]
Same happens with if-else-statements where both have a return.

**Comment by rvalimaki** [comment]
@Jan539 

I might misunderstood, but if you have an if-else statement both having a return, you shouldn't have. Since if your "if" already returns, there is no need for "else". Rest of the code is "else", and thus "else" should be omitted.

**Comment by Matmo10** [comment]
I also get issues with this rule in the following scenario:

```
switch(someVar) {
   case 'whatever':
      return 10;

   case 'something':
      return 20;
// .... etc
}
```

It still expects a break despite the return statement being present. Is this bad form?

**Comment by paynoattn** [comment]
@rvalimaki 

if-else conditionals also fail this lint even if only one has a return. such as the following conditional: 

```
...
case 'myCase':
        if (typeof action.payload !== 'number') {
          throw new Error('cannot access id expect from number');
        } else {
          return action.payload;
        }
```
